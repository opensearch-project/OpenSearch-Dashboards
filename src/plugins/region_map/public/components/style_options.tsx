/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
import { i18n } from '@osd/i18n';
import { FormattedMessage } from '@osd/i18n/react';
import { VisOptionsProps } from 'src/plugins/vis_default_editor/public';
import { IServiceSettings } from '../../../maps_legacy/public';
import { NumberInputOption, SelectOption } from '../../../charts/public';
import { RegionMapVisParams } from '../../../maps_legacy/public';

export type StyleOptionsProps = {
  getServiceSettings: () => Promise<IServiceSettings>;
} & VisOptionsProps<RegionMapVisParams>;

function StyleOptions(props: StyleOptionsProps) {
  const { stateParams, vis, setValue } = props;
  return (
    <EuiPanel paddingSize="s">
      <EuiTitle size="xs" id="styleSettingTitleId">
        <h2>
          <FormattedMessage
            id="regionMap.visParams.styleSettingsLabel"
            defaultMessage="Style settings"
          />
        </h2>
      </EuiTitle>
      <EuiSpacer size="s" />

      <SelectOption
        label={i18n.translate('regionMap.visParams.colorSchemaLabel', {
          defaultMessage: 'Color schema',
        })}
        options={vis.type.editorConfig.collections.colorSchemas}
        paramName="colorSchema"
        value={stateParams.colorSchema}
        setValue={setValue}
        id="colorSchemaId"
      />

      <NumberInputOption
        label={i18n.translate('regionMap.visParams.outlineWeightLabel', {
          defaultMessage: 'Border thickness',
        })}
        min={0}
        paramName="outlineWeight"
        value={stateParams.outlineWeight}
        setValue={setValue}
        id="borderThicknessId"
      />
    </EuiPanel>
  );
}

export { StyleOptions };
