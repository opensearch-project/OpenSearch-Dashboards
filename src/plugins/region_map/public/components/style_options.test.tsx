/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import * as React from 'react';
import { StyleOptions } from './style_options';
import renderer, { act } from 'react-test-renderer';

describe('style_options', () => {
  it('renders the Style options comprising of style settings, color schema and border thickness', async () => {
    const props = jest.mock;
    const vis = {
      type: {
        editorConfig: {
          collections: {
            colorSchemas: [],
          },
        },
      },
    };
    const stateParams = {
      colorSchema: {},
      outlineWeight: {},
    };

    let tree;
    await act(async () => {
      tree = renderer.create(<StyleOptions stateParams={stateParams} vis={vis} {...props} />);
    });
    expect(tree.toJSON().children[0].props.id).toBe('styleSettingTitleId');
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
