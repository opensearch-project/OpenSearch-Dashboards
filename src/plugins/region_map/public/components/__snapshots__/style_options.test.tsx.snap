// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style_options renders the Style options comprising of style settings, color schema and border thickness 1`] = `
<div
  className="euiPanel euiPanel--paddingSmall euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
>
  <h2
    className="euiTitle euiTitle--xsmall"
    id="styleSettingTitleId"
  >
    <span>
      Style settings
    </span>
  </h2>
  <div
    className="euiSpacer euiSpacer--s"
  />
  <div
    className="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
    id="colorSchemaId-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        className="euiFormLabel euiFormRow__label"
        htmlFor="colorSchemaId"
      >
        Color schema
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--fullWidth euiFormControlLayout--compressed"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <select
            className="euiSelect euiSelect--fullWidth euiSelect--compressed"
            id="colorSchemaId"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseUp={[Function]}
            value={Object {}}
          >
            <option
              disabled={true}
              hidden={true}
              value="EMPTY_VALUE"
            >
              
            </option>
          </select>
          <div
            className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
          >
            <span
              className="euiFormControlLayoutCustomIcon"
            >
              <span
                aria-hidden="true"
                className="euiFormControlLayoutCustomIcon__icon"
                data-euiicon-type="arrowDown"
                size="s"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
    id="generated-id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        className="euiFormLabel euiFormRow__label"
        htmlFor="generated-id"
      >
        Border thickness
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--fullWidth euiFormControlLayout--compressed"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            className="euiFieldNumber euiFieldNumber--fullWidth euiFieldNumber--compressed"
            id="generated-id"
            min={0}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="number"
            value={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
